Sources/SMTPKitten/SMTPMessage.swift

```diff
@@ -43,3 +43,28 @@ public enum SMTPResponseCode: Int {
     case startMailInput = 354
     case commandNotRecognized = 502
 }
+
+/// SMTP Authentication method.
+public enum SMTPAuthMethod: String, CaseIterable {
```

Comment:
I can add this myself, but  feel that this should be a `struct`, in case we add more auth methods

Reply:
We shouldn't use a `public enum` for `SMTPAuthMethod`, because it doesn't allow us to expand the auth methods in the future. Adding this would force us to "freeze" the enums cases, so we'd be stuck supporting _only_ these auth methods.

The solution is to make a `public struct` around the same enum, that's instead an `internal enum`. By adding `static let`'s for each enum case, we can have the same API but without worrying about API compatibility.

Reply:
See this PR for a related implementation: https://github.com/hummingbird-project/hummingbird/pull/361